src/rgw/rgw_rest_s3.h

```diff
@@ -689,7 +689,67 @@ class RGWLDAPTokenExtractor : public RGWTokenBasedAuthEngine::Extractor {
   std::string get_token() const override;
 };
 
-class S3AuthFactory : public RGWRemoteAuthApplier::Factory {
+class RGWS3V2TokenBasedAuthEngine : public RGWTokenBasedAuthEngine {
```

Comment:
@pritha-srivastava: I'm not sure whether deriving from `RGWTokenBasedAuthEngine` is suitable way for AWS auth. The class is targeting those authenticators which are token-based. OpenStack Identity API and Kerberos might be seen as typical examples. It's worth to add that under *token* I mean a short-living string that can be used multiple times without alternations. A [comment on StackOverflow](http://stackoverflow.com/a/1592572) might be helpful as well.

AWSv2 isn't token-based in this sense. Instead of providing a constant string, client needs to send `AWSAccessKeyId` and `Signature` according the Amazon's terminology. `Signature` is varying with each request.

The second solution proposed by you seems to be the good one. However, I wouldn't name the class with anything having *Token* inside. Also the extractor concept still could be useful. IIRC the AWS auth stuff could be extracted either from HTTP headers or from query string.